src/site/en/blog/2020-07-09-no-to-routers/index.md

```diff
@@ -0,0 +1,246 @@
+
+Yes, you guessed it correctly: virtual routing cannot be implemented because routers assume you only have plain mapping (see #2).
+
+# What should we do then?
+
+All this sounds awful. What should we do to solve our routing problem in a simple, maintainable and scalable way without going crazy in the mean time? I'm glad you asked.
+
+1. You still need to deserialise state from path and serialise it back. There are many libraries that do it just fine, you can check, for instance: [path-to-regexp](https://github.com/pillarjs/path-to-regexp) (used by react-router and page.js), [path-parser](https://github.com/troch/path-parser) (used by router5) or [route-parser](https://github.com/rcs/route-parser). You also need to put that state into browser history (or sometimes not), and there is [history.js](https://github.com/browserstate/history.js/) lib for you. But you should probably just use [Browser History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API) – it is in a good and consistent state across all major browsers right now.
+2. You need to add asynchronous dependencies check which consists of additional script loading and data requests. Usually, this is what your controller is already handling: redux-thunk, redux-saga, mobx, effector – whatever you prefer. If it handles your async logic, it could handle your async dependencies management.
+3. While transition is happening you need to know some meta info about it. It's always different, so you can just save whatever structure you prefer to your state manager. In most cases, the basics you would like to know are: is transition finished, what the current screen is and what is the next screen.
+But you could also store: loading progress, all intermediary transitions (or redirects), additional context to help with transition, scroll position, cause of transition etc.
+Your state manager can store all that data with ease and also keep it accessible for the business logic mentioned above.
+4. You need to render the screen. In a simple case, you need to choose a component by name and render it. In a more complex case, you could choose a component and all the modal screens depending on your business logic. Anyway, that's the only reason for your view library to exist, so use it!
+
+That's it. You got the path, you got dependencies for it, you stored the meta info, you finally rendered the page. **No router was harmed along the way.**
+
+# Conclusion
+
+Routing is extremely hard to build in an abstract way, because of its high coupling with business logic and libraries used to define business logic.
```

Comment:
Мне кажется, абстракция здесь не совсем корректный термин, может лучше так?

```suggestion
It is tough to build a generalised routing solution because of its high coupling with business logic and libraries used to define business logic.
```